plugins/meta/route-override/route-override.go

```diff
@@ -0,0 +1,400 @@
+		for _, nlroute := range routes {
+			if nlroute.Dst != nil &&
+				nlroute.Dst.IP.Equal(route.Dst.IP) &&
+				nlroute.Dst.Mask.String() == route.Dst.Mask.String() {
+				err = netlink.RouteDel(&nlroute)
+			}
+		}
+	} else {
+		for _, netif := range res.Interfaces {
+			if netif.Sandbox != "" {
+				link, _ := netlink.LinkByName(netif.Name)
+				routes, _ := netlink.RouteList(link, netlink.FAMILY_ALL)
+				for _, nlroute := range routes {
+					if nlroute.Dst != nil &&
+						nlroute.Dst.IP.Equal(route.Dst.IP) &&
+						nlroute.Dst.Mask.String() == route.Dst.Mask.String() {
+						err = netlink.RouteDel(&nlroute)
+					}
+				}
+			}
+		}
+	}
+
+	return err
+}
+
+func addRoute(dev netlink.Link, route *types.Route) error {
+	return netlink.RouteAdd(&netlink.Route{
+		LinkIndex: dev.Attrs().Index,
+		Scope:     netlink.SCOPE_UNIVERSE,
+		Dst:       &route.Dst,
+		Gw:        route.GW,
+	})
+}
+
+func processRoutes(netnsname string, conf *RouteOverrideConfig) (*current.Result, error) {
+	netns, err := ns.GetNS(netnsname)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open netns %q: %v", netns, err)
+	}
+	defer netns.Close()
+
+	res, err := current.NewResultFromResult(conf.PrevResult)
+	if err != nil {
+		return nil, fmt.Errorf("could not convert result to current version: %v", err)
+	}
+
+	if conf.FlushGateway {
+		// add "0.0.0.0/0" into delRoute to remove it from routing table/result
+		_, gwRoute, _ := net.ParseCIDR("0.0.0.0/0")
+		conf.DelRoutes = append(conf.DelRoutes, &types.Route{Dst: *gwRoute})
+		_, gwRoute, _ = net.ParseCIDR("::/0")
+		conf.DelRoutes = append(conf.DelRoutes, &types.Route{Dst: *gwRoute})
+
+		// delete given gateway address
+		for _, ips := range res.IPs {
+			if ips.Version == "6" {
+				ips.Gateway = net.IPv6zero
+			} else {
+				ips.Gateway = net.IPv4zero
+			}
+		}
+	}
+
+	newRoutes := []*types.Route{}
+	err = netns.Do(func(_ ns.NetNS) error {
+		// Flush route if required
+		if !conf.FlushRoutes {
+		NEXT:
+			for _, route := range res.Routes {
+				for _, delroute := range conf.DelRoutes {
+					if route.Dst.IP.Equal(delroute.Dst.IP) &&
+						bytes.Equal(route.Dst.Mask, delroute.Dst.Mask) {
+						err = deleteRoute(delroute, res)
+						if err != nil {
+							fmt.Fprintf(os.Stderr, "failed to delte route %v: %v", delroute, err)
+						}
+						continue NEXT
+					}
+
+				}
+				newRoutes = append(newRoutes, route)
+			}
+		} else {
+			deleteAllRoutes(res)
+		}
+
+		if conf.FlushGateway {
+			deleteGWRoute(res)
+		}
+
+		// Get container IF name
+		var containerIFName string
+		for _, i := range res.Interfaces {
+			if i.Sandbox != "" {
+				containerIFName = i.Name
+				break
+			}
+		}
+		// Add route
+		dev, _ := netlink.LinkByName(containerIFName)
+		for _, route := range conf.AddRoutes {
+			newRoutes = append(newRoutes, route)
+			if err := addRoute(dev, route); err != nil {
+				fmt.Fprintf(os.Stderr, "failed to add route: %v: %v", route, err)
+			}
+		}
```

Comment:
Above procedure is incorrect.
You should validate on which already configured interface is configured a subnet, in which the particular gw can be found. Using first found interface will lead to issues.
This also has to be covered with an integration test.

Reply:
Let me double check that we should do check the gateway is already configured or not in the plugin?
As far as I know, the linux kernel network stack will take care of the validation, hence we do not do that in the plugin side. I also output error message from `netlink.RouteAdd()`.